sdk/trace/trace_test.go

```diff
@@ -801,3 +809,43 @@ func TestCustomStartEndTime(t *testing.T) {
 		t.Errorf("expected end time to be %s, got %s", endTime, got.EndTime)
 	}
 }
+
+func TestWithSpanKind(t *testing.T) {
+	var te testExporter
+	tp, _ := NewProvider(WithSyncer(&te), WithConfig(Config{DefaultSampler: AlwaysSample()}))
+	tr := tp.GetTracer("withSpanKind")
+
+	_, span := tr.Start(context.Background(), "WithoutSpanKind")
+	spanData, err := endSpan(&te, span)
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	if spanData.SpanKind != apitrace.SpanKindInternal {
+		t.Errorf("Default value of Spankind should be Internal: got %+v, want %+v\n", spanData.SpanKind, apitrace.SpanKindInternal)
+	}
+
+	sks := []apitrace.SpanKind{
+		apitrace.SpanKindInternal,
+		apitrace.SpanKindServer,
+		apitrace.SpanKindClient,
+		apitrace.SpanKindProducer,
+		apitrace.SpanKindConsumer,
+	}
+
+	for _, sk := range sks {
+		var te testExporter
+		tp, _ := NewProvider(WithSyncer(&te), WithConfig(Config{DefaultSampler: AlwaysSample()}))
+		tr := tp.GetTracer("withSpanKind")
```

Comment:
Why not reuse the exporter, provider and tracer from the outer scope?

Reply:
Previously, the `endSpan` method created a `testExporter` and used it to process a single span. But with the changes of the `Provider`, it started to receive the `testExporter` as a parameter, and it expected that we got exactly *one* span on the exporter, that why we should not reuse it.

There is a simpler way to do this, just reset the `span` list of the `testExporter`.

Reply:
Thank you. :smile:
I think we should do it, and I'll agree to back this up by filing an issue in the spec repo, sometime this week.